benchmark/NonDet/NQueens.hs

```diff
@@ -45,12 +45,10 @@ addOne n curr = do
 queens :: (Alternative m, Monad m) => Int -> m Board
 queens n = foldl' (>>=) (pure empty) (replicate n (addOne n))
 
-runQueens :: Int -> [Board]
-runQueens = run . runNonDet . queens
-
-benchmark :: Gauge.Benchmark
-benchmark = bgroup "N-queens problem"
-  [ bench "4"  $ whnf runQueens 4
-  , bench "8"  $ whnf runQueens 8
-  , bench "16" $ whnf runQueens 16
+benchmark :: (Alternative m, Monad m) => String -> (m Board -> [Board]) -> Gauge.Benchmark
+benchmark title runQueens = bgroup title
```

Comment:
We pass in the title and handler so that we can run the benchmark group against different carriers.